src/syft/core/node/common/action/function_or_constructor_action.py

```diff
@@ -33,29 +41,144 @@ def __init__(
         #  id on the object directly
         self.id_at_location = id_at_location
 
+    @staticmethod
+    def intersect_keys(
+        left: Union[Dict[VerifyKey, UID], None], right: Dict[VerifyKey, UID]
+    ) -> Dict[VerifyKey, UID]:
+        # FIXME duplicated in run_class_method_action.py
+        # get the intersection of the dict keys, the value is the request_id
+        # if the request_id is different for some reason we still want to keep it,
+        # so only intersect the keys and then copy those over from the main dict
+        # into a new one
+        if left is None:
+            return right
+        intersection = set(left.keys()).intersection(right.keys())
+        # left and right have the same keys
+        return {k: left[k] for k in intersection}
+
     def execute_action(self, node: AbstractNode, verify_key: VerifyKey) -> None:
         method = node.lib_ast(self.path)
 
+        result_read_permissions = None
+
         resolved_args = list()
         for arg in self.args:
-            if isinstance(arg, Pointer):
-                # QUESTION: Where is get_object definied?
-                r_arg = node.store.get_object(id=arg.id_at_location)
-                resolved_args.append(r_arg)
-            else:
-                resolved_args.append(arg)
+            if not isinstance(arg, Pointer):
+                raise ValueError(
+                    f"args attribute of RunFunctionOrConstructorAction should only contain Pointers. "
```

Comment:
Is this true? We will be able to pass in tuples no? (aka - not pointers to tuples)

Reply:
That's correct, I hastily replicated what was done in `RunClassMethodAction`
Do we also want "primary" types as int, etc. to do stuff as `t.repeat(2)` without having to send the `2`?

Reply:
What case do you think I should implement for now? Only the tuple/list argument?